src/server/rpc/procedures/iex-trading/iex-trading.js

```diff
@@ -94,6 +94,35 @@ StockConsumer.dailyPercentChange = function(companySymbol) {
         });
 };
 
+/**
+ * Get historical closing price information about the specified company
+ * @param {String} companySymbol Company stock ticker symbol
+ * @param {String} range Time period (e.g. 1m, 3m, 1y, etc) to retrieve prices for
+ * @returns {Object} Historical price information
+ */
+StockConsumer.historicalClosingPrices = function(companySymbol, range) {
+    return this._requestData({queryString: `/stock/${companySymbol}/chart/${range}`})
+            .then(res => {
+                let output = {};    
+
+                // Convert data to more usable table        
+                for(let i = 0; i < res.length; i++)
+                {
+                    output[res[i]['date']] = res[i]['close'];
```

Comment:
This seems to work currently because `Object.keys` returns the keys in the insertion order in v8. However, it might be best not to rely on this since it is not guaranteed by the JS spec. That is, it might not be bad to change this to something like `return prices.map(price => [price.date, price.close])` where `prices` is the result of `this._requestData` (currently `res`).

Reply:
This is changed in 476149b